linalg/sparsemat.cpp

```diff
@@ -4120,10 +4123,16 @@ SparseMatrix * Add(real_t a, const SparseMatrix & A, real_t b,
    int * C_j;
    real_t * C_data;
 
+   A.HostReadI();
+   A.HostReadJ();
+   A.HostReadData();
    const int *A_i = A.GetI();
    const int *A_j = A.GetJ();
    const real_t *A_data = A.GetData();
```

Comment:
It is simpler to just replace `GetI()` with `HostReadI()`. The same applies to the other arrays and for the `B` matrix below.

Reply:
done in https://github.com/mfem/mfem/pull/4675/commits/6b7b9985e7ecec8cdaa95fab6f63bbe324ef969b